x/hard/keeper/borrow.go

```diff
@@ -113,6 +113,17 @@ func (k Keeper) ValidateBorrow(ctx sdk.Context, borrower sdk.AccAddress, amount
 		return types.ErrBorrowEmptyCoins
 	}
 
+	// The reserve coins aren't available for users to borrow
+	hardMaccCoins := k.supplyKeeper.GetModuleAccount(ctx, types.ModuleName).GetCoins()
+	reserveCoins, foundReserveCoins := k.GetTotalReserves(ctx)
+	if !foundReserveCoins {
+		reserveCoins = sdk.NewCoins()
+	}
+	fundsAvailableToBorrow := hardMaccCoins.Sub(reserveCoins)
```

Comment:
Should be `SafeSub` - there are scenarios where the protocol is insolvent and `reserves > cash`

Reply:
Added `SafeSub` and new error message in https://github.com/Kava-Labs/kava/pull/827/commits/c70fce8d7eeef6705f91c79c936c1f46cad3b110